tests/e2e/manifests/verify_manifest_lists.go

```diff
@@ -459,6 +463,26 @@ func verifyManifestList(manifest, imageName, tag string) error {
 	aList := make([]string, len(archList))
 	// copy into a temp slice.
 	copy(aList, archList)
+	// remove arch's from the list if the k8s version no longer supports them
+	for versionToRemove, archs := range architecturesToRemove {
+		verWithoutPreRelease, err := version.ParseSemantic(fmt.Sprintf("%d.%d.%d", ver.Major(), ver.Minor(), ver.Patch()))
+		if err != nil {
+			return fmt.Errorf("could not parse version %q: %v", ver.String(), err)
+		}
+		if verWithoutPreRelease.AtLeast(version.MustParseSemantic(versionToRemove)) {
```

Comment:
is this correct? reading the logs it seems the arm arch was removed in 1.27 pre-release, but i think with this we will only not check for arm in >= 1.27.0.

Reply:
`Version.AtLeast()` doesnt support evaluating pre releases which is why I created the `verWithoutPreRelease` variable. This way prerelease versions will be removed from semantic versioning so "1.27.0-rc.1" is evaluated as "1.27.0".

Reply:
One last bit of confirmation: https://go.dev/play/p/gLsU3Oesu6C

Reply:
That being said, because of the `AtLeast()` limitation, this code as is would not handle a situation where an arch was removed during a pre-release i.e supported in "1.X.0-rc.1" and removed in "1.X.0-rc.2". I think the chances of such a large change/decision happening between pre-release's is unlikely, but something to be aware of.

Reply:
>  I think the chances of such a large change/decision happening between pre-release's is unlikely, but something to be aware of.

it might happen, as a forgotten task before a release.
can we just use the Compare function with an exact version even if it's pre-release?

Reply:
Awesome recommendation! It looks like that'll work. https://go.dev/play/p/fJF35HOtd53 I'll make the updates.

Reply:
Ok, updated to work with prerelease versions. Thanks!